lib/src/drawing_and_graphics/shadow.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:ui';
+
+import 'package:flutter/widgets.dart';
+
+/// A widget that adds a shadow effect to its child widget.
+class Shadow extends StatelessWidget {
+  /// Creates a Shadow widget with the given properties.
```

Comment:
When there's only a single constructor, I tend to find that docs on the constructor don't help much. For example "Creates a Shadow widget with the given properties" is exactly what the code tell us, without docs.

Given that there's just one constructor, I suggest providing all relevant widget-wide details in the class Dart Doc, and then provide per property clarifications in the property docs.

For example, one thing I would expect to see explained in the class Dart Doc is how a `Shadow` widget applies a shadow to its child. How does `Shadow` know the shape of its child? Are there any further requirements to make this widget work? Etc.

Reply:
Good point. I removed the constructor Dart Doc and improved the class one. I didn't try to explain how Shadow knows the shape of its child, though. That seems like an implementation detail that is unnecessary to know at a high level. Shadows in swift_ui just work no matter what the shape is (unlike in vanilla Flutter), so there is nothing special a developer needs to do.

Reply:
Exposing implementation details in docs is fine, so long as those details are directly related to the applicability of the object.

For example, let's say that I display a bitmap image of an oval with transparency. Then I apply a shadow. In Photoshop, I'll get a shadow around the oval. But in Figma I get a shadow around the rectangle of the whole bitmap. Shadows have internal details that leak out and we should make them clear so that people don't have the wrong expectations.

I assume that the underlying shadow mechanism in SwiftUI is based around some kind of vector graphics protocol, is that right? If you modify a bitmapw ith transparency in SwiftUI with a shadow, does the shadow apply to the image bounds, or the visible bitmap bounds?

Assuming that SwiftUI does the same thing that we do, we should at least document what that restriction is.

Reply:
I understand now. I updated the doc comments to reflect how the shadow works.

I'm not sure if SwiftUI uses a vector graphics protocol or not, but the shadow applies to the visible parts of the image, even for a bitmap with transparency. I tested this with the dash_hello image. Here it is in SwiftUI:

<img width="194" alt="Screenshot 2024-07-25 at 10 08 36" src="https://github.com/user-attachments/assets/35fad44e-4479-491d-9a7c-6bab23421355">

```
Image("dash_hello")
    .resizable()
    .aspectRatio(contentMode: .fit)
    .shadow(color: .gray, radius: 10, x: 10, y: 10)
    .frame(width: 200, height: 200)
```

And this is the Flutter version:

<img width="263" alt="Screenshot 2024-07-25 at 10 08 47" src="https://github.com/user-attachments/assets/11af05b8-474e-403d-b967-041bc48ad687">

```
Frame(
  width: 200,
  height: 200,
  Shadow(
    color: Color(0xFF8E8E93),
    radius: 10.0,
    x: 10.0,
    y: 10.0,
    Image.asset(
      'assets/images/dash_hello/dash_hello.png',
      fit: BoxFit.contain,
    ),
  ),
),
```

The behavior is the same for both.